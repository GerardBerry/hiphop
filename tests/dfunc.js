"use hopscript"

const hh = require("hiphop");

const prg =
      <hh.module>
	<hh.iosignal name="s1" />
	<hh.iosignal name="s2" valued/>
	<hh.parallel>
	  <hh.sequence>
	    ${function(){console.log("--br1--1")}}
	    <hh.emit signal="s2" dfunc=${function() {return this.present.s1}}/>
	    ${function(){console.log("--br1--2")}}
	  </hh.sequence>
	  <hh.sequence>
	    ${function(){console.log("--br2--1")}}
	    <hh.emit signal="s1"/>
	    ${function(){console.log("--br2--2")}}
	  </hh.sequence>
	</hh.parallel>
      </hh.module>

const machine = new hh.ReactiveMachine(prg);

machine.react();
