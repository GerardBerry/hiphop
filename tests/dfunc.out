--br1--1
--br2--1
--br1--2
--br2--2
